include/NovelRT.Interop/Input/NrtInteractionService.h

```diff
@@ -0,0 +1,31 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_INTEROP_INPUT_INTERACTIONSERVICE_H
+#define NOVELRT_INTEROP_INPUT_INTERACTIONSERVICE_H
+
+#include "../Windowing/NrtWindowingService.h"
+#include "../NrtTransform.h"
+#include "../Maths/NrtGeoVector2F.h"
+#include "../NrtInteropUtils.h"
+#include "NrtKeyCode.h"
+#include "NrtKeyStateFrameChangeLog.h"
+#include "NrtBasicInteractionRect.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct InteractionServiceHandle* NrtInteractionService;
+
+NrtInteractionService Nrt_InteractionService_create(const NrtWindowingService windowingService);
+NrtResult Nrt_InteractionService_consumePlayerInput(const NrtInteractionService service);
+NrtResult Nrt_InteractionService_createBasicInteractionRect(const NrtInteractionService service, const NrtTransform transform, int layer, NrtBasicInteractionRect* outputRect);
```

Comment:
int vs int32_t again, man we really have inconsistencies with this don't we lol

Reply:
Anything using layer is an inconsistency with this lol